.github/workflows/ci.yml

```diff
@@ -96,8 +96,21 @@ jobs:
 
     - run: nox --python ${{ matrix.python.action }} -e ${{ matrix.task.nox }} -- --use-wheel dist/*.whl
 
+    - name: Store coverage file
+      uses: actions/upload-artifact@v3
+      with:
+        # We have a single artifact shared for each workflow run.
+        # in this way we should intefere with coverage reports from other PRs
+        # or previous runs for the same PR.
+        name: coverage-${{ github.run_id }}
+        # Is important to keep the unique names for the coverage files
+        # so that when uploaded to the same artifact, they don't overlap.
+        path: .coverage*
```

Comment:
See comments in duplicated section below.

Side note...  I really like cross-platform matrices to avoid this.  `:]`

Reply:
I am missing the comment.

I am happy to see less duplication, but I think is best to have it in separate PR.

Thanks

Reply:
This is copy pasted from the other similar section and contains the same typos that were present there and have been fixed.  They should be fixed here as well.

But, I'm going to go ahead and trigger a run here, cancel it partially completed, rerun the cancelled jobs and see if my concern about using the `github.run_id` seems accurate.